src/toast/tests/ops_sim_gaindrifts.py

```diff
@@ -0,0 +1,348 @@
+        drifter.apply(data)
+        for obs, old in zip(data.obs, old_data):
+            telescope = obs.telescope.uid
+            focalplane = obs.telescope.focalplane
+            obsindx = obs.uid
+
+            key1 = (
+                drifter.realization * 4294967296 + telescope * 65536 + drifter.component
+            )
+            counter1 = 0
+            counter2 = 0
+            for det in obs.local_detectors:
+                detindx = focalplane[det]["uid"]
+                key2 = obsindx * 4294967296 + detindx
+                rngdata = rng.random(
+                    1,
+                    sampler="gaussian",
+                    key=(key1, key2),
+                    counter=(counter1, counter2),
+                )
+                gf2 = 1 + rngdata[0] * drifter.sigma_drift
+                gf1 = (obs.detdata[key][det] / old[det])[-1]
+                # assert whether the two values gf2 and gf1  are the same
+                # within 1sigma of the distribution
+                np.testing.assert_almost_equal(
+                    gf1, gf2, decimal=np.log10(drifter.sigma_drift) - 1
+                )
+
+
+    def test_thermal_drift(self):
+        # Create a fake satellite data set for testing
+        data = create_satellite_data_big(
+            self.comm,pixel_per_process=7
+        )
+        # Create a noise model from focalplane detector properties
+        default_model = ops.DefaultNoiseModel()
+        default_model.apply(data)
+        # make a simple pointing matrix
+        detpointing = ops.PointingDetectorSimple()
+        pointing = ops.PointingHealpix(
+            nside=16,
+            nest=False,
+            mode="I",
+            detector_pointing=detpointing,
+        )
+        # Generate timestreams
+        key = "signal"
+        sim_dipole = ops.SimDipole(det_data=key, mode="solar", coord="G")
+        sim_dipole.apply(data)
+
+        # Build the covariance and hits
+        cov_and_hits = ops.CovarianceAndHits(
+            pixel_dist="pixel_dist",
+            pointing=pointing,
+            noise_model=default_model.noise_model,
+            rcond_threshold=1.0e-6,
+            sync_type="alltoallv",
+        )
+        cov_and_hits.apply(data)
+
+        # Set up binned map
+
+        binner1 = ops.BinMap(
+            pixel_dist="pixel_dist",
+            covariance=cov_and_hits.covariance,
+            det_data=key,
+            pointing=pointing,
+            noise_model=default_model.noise_model,
+            sync_type="alltoallv",
+        )
+        binner1.apply(data)
+        map1_path = os.path.join(self.outdir, "toast_bin1_drift.fits")
+        write_healpix_fits(data[binner1.binned], map1_path, nest=False)
+
+        # inject gain drift  w/ common mode
+
+        drifter = ops.GainDrifter(
+            det_data=key,
+            drift_mode="thermal_drift",
+            detector_mismatch=0.7,
+        )
+        drifter.apply(data)
+        binner2 = ops.BinMap(
```

Comment:
This operator is using all the same traits as `binner1` (including the key for the output map in data), so you can just apply the original binner again.